agent/consul/state/acl.go

```diff
@@ -536,6 +537,46 @@ func (s *Store) ACLTokenListUpgradeable(max int) (structs.ACLTokens, <-chan stru
 	return tokens, iter.WatchCh(), nil
 }
 
+func (s *Store) aclTokenUnlinkPolicyTxn(tx *memdb.Txn, idx uint64, policyID string) (bool, error) {
+	iter, err := tx.Get("acl-tokens", "policies", policyID)
+	if err != nil {
+		return false, err
+	}
+
+	updates := false
+
+	for raw := iter.Next(); raw != nil; raw = iter.Next() {
+		original := raw.(*structs.ACLToken)
+		token := &structs.ACLToken{
+			AccessorID:  original.AccessorID,
+			SecretID:    original.SecretID,
+			Description: original.Description,
+			Type:        original.Type,
+			Rules:       original.Rules,
+			Local:       original.Local,
```

Comment:
Should the CreateTime get copied here as well?

Reply:
`CreateTime` is being treated like the other metadata `Hash` and `RaftIndex`. These are set/copied/updated in `aclTokenSetInternal`. I could have set it here too but it would be an unnecessary copy that we will do unconditionally within the other function.